scripts/list-build-dependencies.sh

```diff
@@ -0,0 +1,298 @@
+#!/bin/bash
```

Comment:
I imagined this file would be much smaller! :)

I suspect a file, which would look sort-of like this:

    #!/bin/bash
    
    case "$1" in
    	macos)
    		echo "coreutils autogen autoconf automake pkg-config libpng sdl sdl_net opusfile speexdsp" ;;
    	ubuntu)
    		echo "package1 package2 (...)" ;;
    	ubuntu-32)
    		echo "packageX packageY (...)" ;;
    	fedora)
    		echo "packageN packageM (...)" ;;
    	msys2-64)
    		echo "(...)" ;;
    	msys2-32)
    		echo "(...)" ;;
    	vcpkg)
    		echo "(...)" ;;
    esac


could probably fulfil all our requirements here? (of course this is not exact code, just an idea) I think we could drop compilers from being included with the build dependencies (those are usually not debatable for packaging in distros anyway, and could be easily hardcoded in the .yml files).

Reply:
Yes! I think we can rip out all functions and boil it down to something elegant like that, and still handle stuff like --bit-depth 32 vs. 64 with only a couple if/elses. 
For now though, because it works and has been test maybe we could do this as an "version 2" optimization, and open an issue against it after merging?

Reply:
OK :) I just checked and there are no shellcheck warnings in this file any more for this review.